### PR TITLE
feat: [CDS-36402]: Filestore UI

### PR DESCRIPTION
**Pull request by radujomir**
##### Summary:

Filestore UI

##### Jira Links:

(https://harness.atlassian.net/browse/CDS-36425)

##### Screenshots:
Screenshot 2022-06-09 at 16 55 26

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by azharness** [comment]
retrigger titlecheck

**Comment by kapilkumawat86** [comment]
retrigger jest

**Comment by kapilkumawat86** [comment]
trigger rebase

File: src/modules/28-filestore/images/closed-folder.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M4 4H10L12 7H20V20H4V4Z" stroke="#0278D5" stroke-linecap="round" stroke-linejoin="round"/>
```
Comment (vkbansal) [inline src/modules/28-filestore/images/closed-folder.svg:2]: Shouldn't these icons be in uicore?

**Review by vkbansal — APPROVED** [review]

File: src/modules/28-filestore/common/FileStorePopover/FileStorePopover.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          resource: {
+            resourceType: ResourceType.FILE
+          }
+        }}
+        onClick={e => {
+          e.stopPropagation()
+          setMenuOpen(true)
+        }}
+        rightIcon="chevron-down"
+        disabled={false}
```
Comment (bhavya-sinha) [inline src/modules/28-filestore/common/FileStorePopover/FileStorePopover.tsx:71]: disabled is false by default

File: src/modules/28-filestore/common/FileStorePopover/FileStorePopover.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        id="newFileBtn"
+        data-testid="newFileButton"
+      />
+      <Menu>
+        {items.length &&
+          items.map((item: FileStorePopoverItem, index: number) => {
+            const { label, onClick, actionType, identifier } = item
+            return (
+              <RbacMenuItem
+                key={index}
```
Comment (bhavya-sinha) [inline src/modules/28-filestore/common/FileStorePopover/FileStorePopover.tsx:82]: Better to use identifier as key

File: src/modules/28-filestore/common/ModalComponents/ModalComponents.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import React from 'react'
+
+import { Layout, Button, ButtonVariation } from '@harness/uicore'
+
+interface FooterRendererProps {
+  onSubmit?: () => void
+  onCancel: () => void
+  confirmText: string
+  cancelText: string
+  type?: 'submit' | undefined
```
Comment (bhavya-sinha) [inline src/modules/28-filestore/common/ModalComponents/ModalComponents.tsx:17]: undefined not needed with optional param

File: src/modules/28-filestore/common/useFileStoreScope/useFileStoreScope.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    if (scope === Scope.ORG) {
+      return {
+        accountIdentifier: accountId,
+        orgIdentifier
+      }
+    }
+    if (scope === Scope.PROJECT) {
+      return {
+        accountIdentifier: accountId,
```
Comment (bhavya-sinha) [inline src/modules/28-filestore/common/useFileStoreScope/useFileStoreScope.tsx:43]: Better to use switch here

Comment (litvinovstas) [reply src/modules/28-filestore/common/useFileStoreScope/useFileStoreScope.tsx]: changed

File: src/modules/10-common/components/Filter/Constants.ts [hunk 6]
```diff
@@ -13,8 +13,8 @@ export interface FilterInterface {
   filterVisibility?: FilterDTO['filterVisibility']
   identifier: string
   filterProperties?: any
-  orgIdentifier?: string
-  projectIdentifier?: string
+  orgIdentifier?: string | undefined
```
[the remark this answers is not in the document]
Comment (litvinovstas) [reply src/modules/10-common/components/Filter/Constants.ts]: resolved

**Review by vkbansal — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**